### DISCUSSION TITLE
The `Nonempty` Option Feels Wrong

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#4344] [Q&A] [answered]
Hi

I have these two collections:

- `users`
- `tiers`

The `users` collection has a field called `retainedTiers` (a relation of `tiers`).

For `retainedTiers`:

- When `Nonempty` is disabled, it means the field can be `undefined`, and when we create a user, we don't need to pass in `retainedTiers`. It also means the field is not returned when reading/fetching the collection.
- When `Nonempty` is enabled, the value can't be an empty array. This means the field must have at least 1 item inside it.

The problems:

- It seems like when `Nonempty` is enabled, the "Min select" option is ignored. No matter if it's not set or if it's set to 0.
- When `Nonempty` is enabled, I can't delete the tier that is used by a user.
- When `Nonempty` is not enabled, I expect to get the property back when I fetch the collection.

Here is a video:

https://github.com/pocketbase/pocketbase/assets/44144724/e08a6071-c539-4c8d-8be6-30e942110a74

**Comment by ganigeorgiev** [comment, marked as answer]
> When Nonempty is disabled, it means the field can be undefined

No, it doesn't mean that. For all fields, with exception to the `json` field type, we apply the zero-default of the value. See the first value shown in the table in https://pocketbase.io/docs/collections/#overview.

There should be also a tooltip next to the "Nonempty" toggle explaining what does it means.

**Reply by babakfp** [reply]
Does that mean the type below generated by https://github.com/patmood/pocketbase-typegen is incorrect?

```ts
export type UsersRecord = {
    retainedTiers?: RecordIdString[]
    //      This ^ means `retainedTiers` is expected to be `undefined`.
}
```

... and it should be this instead?

```ts
export type UsersRecord = {
    retainedTiers: RecordIdString[]
}
```

**Reply by ganigeorgiev** [reply]
I guess it depends on your context. All collection fields (with exception to the `json` field type) will have a default value as shown in the [Overview section](https://pocketbase.io/docs/collections/#overview).
But if you use for example the `?fields` query parameter and decide to skip some of the fields then they will not be available in the response.
